third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -906,9 +918,9 @@ impl LoaderV1 {
 
         while let Some((addr, name, allow_loading_failure)) = stack.pop() {
             // Load and deserialize the module only if it has not been cached by the loader.
-            // Otherwise this will cause a significant regression in performance.
+            // Otherwise, this will cause a significant regression in performance.
             let (module, size) = match module_store.module_at_by_ref(addr, name) {
-                Some(module) => (module.module.clone(), module.size),
+                Some((module, size)) => (module.module.clone(), size),
```

Comment:
previous interface seems better since module still contains size? (and otherwise there is a chance to return some other size)?

Reply:
My goal is to make size completely oblivious of module representation. So if you need size, you can fetch it from storage. Now a fair question is if we already have a `Module`, can we just extract size avoiding possibly expensive callback to the executor? Here I have only moved it out of the module but it still lives in module cache. 

Do you want to put it back? I think one other reason why I did it was also to avoid the need for passing size when creating `Module` from `Arc<CompiledModule>` (because you need it there as well, and in the end need `ModuleStorage` to fetch it for you).

Reply:
I think fetching there is okay, I'd like to avoid callbacks to the executor.

Reply:
Size is back :)